### PR TITLE
[RORO-5128] upgrade to scala 3

### PR DESCRIPTION
**Pull request by tianjianfeng**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,7 +6,7 @@ lazy val microservice = Project("gvms-haulier-status", file("."))
   .enablePlugins(play.sbt.PlayScala, SbtDistributablesPlugin)
   .settings(
     majorVersion := 0,
-    scalaVersion := "2.13.12",
+    scalaVersion := "3.4.2",
```
Comment (sureshhmrc) [inline build.sbt]: can we all stick to one version going forward 3.5.0 unless i missed something ?

Comment (tianjianfeng) [reply build.sbt]: As Ben suggested, we can leave it for now until the next dev ticket comes up.

Comment (tipperba) [reply build.sbt]: I meant the tickets which are in Done (or already merged) don't need to be changed right now to 3.5.0. There's now downside to changing this one as it's still on branch.

Comment (tianjianfeng) [reply build.sbt]: upgrade to 3.5.0 now.

**Review by elabeca — DISMISSED** [review]

**Review by tipperba — DISMISSED** [review]

**Review by tipperba — APPROVED** [review]

**Pull request merged**